File: contest_sites/timus/vol1/1014.py
#!/usr/bin/python3
# JUDGE_ID: 41750CH

import sys

def divide_by_d(n, d):
    count = 0
    r = n % d
    while r == 0 and n > 1:
        count += 1
        n = n // d
        r = n % d
    return (n, count)

def calculate(n):
    if n == 0:
        print(10)
    elif n >= 1 and n <= 9:
        print(n)
    else:
        count_of_divisions = [0 for i in range(10)]
        nn = n

        i = 9
        while i >= 2:
            nn, c = divide_by_d(nn, i)
            count_of_divisions[i] = c
            i -= 1

        if not nn == 1:
            print(-1)
        else:
            #v = 1
            number = ''
            for i in range(2, 10):
                number += str(i) * count_of_divisions[i]
                #v *= i ** count_of_divisions[i]
            print(number)
            # if v != n:
            #     print(v)
            #     print(n)
            #     sys.exit("Value not equal")



#for i in range(10**9+1):
#    print(i, ' : ', end ='')


n = int(input())
calculate(n)

File: contest_sites/timus/vol1/1014.cpp
#include <iostream>
#include <cmath>
#include <vector>
using namespace std;

vector<int> divide_by_d(int n, int d)
{
  int count = 0;
  int r = n % d;
  while (r == 0 && n > 1){
    count++;
    n /= d;
    r = n % d;
  }
  vector<int> rvalue {n, count};
  return rvalue;
}


int calculate(int n)
{
  if (n == 0)
    cout << 10 << endl;
  else if (n>=1 && n<=9)
    cout << n << endl;
  else{
    vector<int> count_of_divisions (10, 0);
    int nn = n;
    for(int i=9; i>=2; --i){
      vector<int> value = divide_by_d(nn, i);
      nn = value[0];
      count_of_divisions[i] = value[1];
    }

    if (nn == 1){
      for(int i=2; i<10; i++)
        for(int j=0; j<count_of_divisions[i]; j++)
          cout << i;
      cout << endl;
    }
    else {
      cout << -1 << endl;
    }
  }
  return 0;
}

int main()
{
  int n;
  cin >> n;
  calculate(n);
  return 0;
}
